lib/remap-parser/src/parser.lalrpop

```diff
@@ -0,0 +1,501 @@
+
+ArithmeticExpr: Node<Expr> = {
+    Sp<ErrorCoalesce>,
+};
+
+ErrorCoalesce: Expr = {
+    Op<ErrorCoalesce, "??", Logical>,
+    Logical,
+};
+
+Logical: Expr = {
+    Op<Logical, "||", Equal>,
+    Op<Logical, "&&", Equal>,
+    Equal,
+};
+
+Equal: Expr = {
+    Op<Equal, "!=", Compare>,
+    Op<Equal, "==", Compare>,
+    Compare,
+};
+
+Compare: Expr = {
+    Op<Compare, ">=", Add>,
+    Op<Compare, ">", Add>,
+    Op<Compare, "<=", Add>,
+    Op<Compare, "<", Add>,
+    Add,
+};
+
+Add: Expr = {
+    Op<Add, "+", Factor>,
+    Op<Add, "-", Factor>,
+    Factor,
+};
+
+Factor: Expr = {
+    Op<Factor, "%", Term>,
+    Op<Factor, "*", Term>,
+    Op<Factor, "/", Term>,
+    Term,
+};
+
+Term: Expr = {
+    Sp<Literal> => Expr::Literal(<>),
+    Sp<Container> => Expr::Container(<>),
+    Sp<Query> => Expr::Query(<>),
+    Sp<FunctionCall> => Expr::FunctionCall(<>),
+    Sp<Ident> => Expr::Variable(<>),
+};
+
+// -----------------------------------------------------------------------------
+// query
+// -----------------------------------------------------------------------------
+
+pub Query: Query = {
+    LQuery <Sp<".">> RQuery => {
+        let span = <>.span();
+        let target = Node::new(span, QueryTarget::External);
+        let path = Node::new(span, Path(vec![]));
+
+        Query { target, path }
+    },
+    LQuery <target: Sp<QueryTarget>> <path: Sp<Path>> RQuery => Query { target, path },
+};
+
+QueryTarget: QueryTarget = {
+    Ident => QueryTarget::Internal(<>),
+    "." => QueryTarget::External,
+    FunctionCall => QueryTarget::FunctionCall(<>),
+    Container => QueryTarget::Container(<>),
+};
+
+// -----------------------------------------------------------------------------
+// path
+// -----------------------------------------------------------------------------
+
+Path: Path = PathSegment+ => Path(<>);
+
+PathSegment: PathSegment = {
+    "."? <Field> => PathSegment::Field(<>),
+    "[" <Integer> "]" => PathSegment::Index(<>),
+    "."? "(" <v:(<Field> "|")+> <e:Field> ")" => {
+        let mut v = v;
+        v.push(e);
+        PathSegment::Coalesce(v)
+    },
+};
+
+pub Field: Field = {
+    AnyIdent => Field::Regular(<>),
+    String => Field::Quoted(<>),
+};
+
+// -----------------------------------------------------------------------------
+// function call
+// -----------------------------------------------------------------------------
+
+FunctionCall: FunctionCall = {
+    <ident: Sp<"function call">> <abort_on_error: "!"?> "("
+        NonterminalNewline*
+        <arguments: CommaMultiline<Sp<FunctionArgument>>?>
+    ")" => {
+        let ident = ident.map(|s| Ident(s.to_owned()));
+        let abort_on_error = abort_on_error.is_some();
+        let arguments = arguments.unwrap_or_default();
+
+        FunctionCall { ident, abort_on_error, arguments }
+    },
+};
+
+#[inline]
+FunctionArgument: FunctionArgument = {
+    <ident: (<Sp<AnyIdent>> ":")?> <expr: ArithmeticExpr> => FunctionArgument { <> },
```

Comment:
Note that LR(1) parsers aren't allowed to be ambiguous, so this isn't allowed:

```rust
foo(ok, err)
foo(ok, err = ...)
```

Both cases would be parsed the same, but the first is two positional arguments, the second is one error-assignment expression for the first positional argument.

To work around this _for now_, using error-assignments in cases like these requires you to wrap it in parenthesis:

```rust
foo((ok, err = ...), second_arg)
```

single assignments still work without:

```rust
foo(ok = ..., second_arg)
```